eox_nelp/signals/receivers.py

```diff
@@ -112,3 +115,73 @@ def certificate_publisher(certificate, metadata, **kwargs):  # pylint: disable=u
             certificate.user.pii.username,
             certificate.course.course_key,
         )
+
+
+def enrollment_publisher(instance, **kwargs):  # pylint: disable=unused-argument
+    """
+    Receiver that is connected to the course enrollment post_save signal and this will generate certificate
+    data to publish it to the external service. That behavior is controlled by the following settings:
+
+    - ENABLE_CERTIFICATE_PUBLISHER<boolean>: If this is true the receiver will publish the certificate data,
+    default is False.
+    - CERTIFICATE_PUBLISHER_VALID_MODES<list[string]>: List of valid modes, default ['no-id-professional']
+
+    Note: This keeps the same certificate receiver settings since this will create an external certificate at
+    the beginning of the course, then the certificate receiver will update the grade.
+
+    Args:
+        instance<CourseEnrollment>: This an instance of the model CourseEnrollment.
+    """
+    if not getattr(settings, "ENABLE_CERTIFICATE_PUBLISHER", False):
+        return
+
+    default_modes = [
+        "no-id-professional",
+    ]
+    valid_modes = getattr(settings, "CERTIFICATE_PUBLISHER_VALID_MODES", default_modes)
+
+    if instance.mode in valid_modes:
+        LOGGER.info(
+            "The %s enrollment associated with the user <%s> and course <%s> has been already generated "
+            "and its data will be sent to the NELC certificate service.",
+            instance.mode,
+            instance.user.username,
+            instance.course_id,
+        )
+        time = instance.created
+        user = instance.user
+        course_grade = CourseGradeFactory().read(user, course_key=instance.course_id)
+        certificate = CertificateData(
+            user=UserData(
+                pii=UserPersonalData(
+                    username=user.username,
+                    email=user.email,
+                    name=user.profile.name,
+                ),
+                id=user.id,
+                is_active=user.is_active,
+            ),
+            course=CourseData(
+                course_key=instance.course_id,
+            ),
+            mode=instance.mode,
+            grade=course_grade.percent,
+            current_status='downloadable' if course_grade.passed else 'not-passing',
+            download_url='',
+            name='',
+        )
+
+        create_external_certificate.delay(
+            external_certificate_data=_generate_external_certificate_data(
```

Comment:
Here I found a problem if I create an enrollment from scratch.
Maybe the certificate doesn't exist and resolves
![Peek 2023-08-31 15-15](https://github.com/eduNEXT/eox-nelp/assets/51926076/d9469b06-cd76-4e65-b32a-b5a41522c7a6)
 with 500.

Reply:
Good catch, the good new is that you found a line that is not necessary